packages/vue-instantsearch/src/util/vue-compat/index-vue3.js

```diff
@@ -28,7 +29,7 @@ export function renderCompat(fn) {
       delete flatProps.attrs;
       delete flatProps.props;
       delete flatProps.scopedSlots;
-      delete flatProps.on;
+      Object.keys(defaultedOnProp).forEach((key) => delete flatProps.on[key]);
```

Comment:
Removing `on` entirely was deleting the passed [`on` prop ](https://www.algolia.com/doc/api-reference/widgets/toggle-refinement/vue/#widget-param-on), resulting in it being defaulted all the time.